File: algo/bj2747.py
memo = {1:1, 2:1}
def fibo(num):
	if num <  2:
		return num
	if num not in memo:
		memo[num] = fibo(num-1)+fibo(num-2)
	return memo[num]

import sys

data = sys.stdin.readline().split()
print(fibo(int(data[0])))

File: algo/bj2747.cpp
#include<stdio.h>

int fibo(int);

int main()
{
	int num,result;
	scanf("%d", &num);
	result = fibo(num);
	printf("%d\n",result);
	return 0;
}

int fibo(int num)
{
	if(num <2)
	{
		return num;
	}
	return fibo(num-1)+fibo(num-2);
}
